ql/src/test/org/apache/hadoop/hive/ql/io/orc/TestOrcFile.java

```diff
@@ -1818,8 +1818,9 @@ long getTotalMemoryPool() {
     }
 
```

Comment:
It will be good to add tests that uses WriterOptions.setSchema() to specify the types instead of OI. I see that you have fixed bunch of OI casts that would have cause NPE otherwise (if TypeDescription is set through writer).